sources/api/apiclient/src/network.rs

```diff
@@ -0,0 +1,258 @@
+pub type Result<T> = std::result::Result<T, error::Error>;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use base64::{engine, Engine};
+
+    #[tokio::test]
+    async fn test_get_content_base64() {
+        // Given a valid TOML network configuration encoded in base64
+        let test_content = "version = 2\n\n[eth0]\ndhcp4 = true\n";
+        let encoded = engine::general_purpose::STANDARD.encode(test_content.as_bytes());
+        let base64_uri = format!("base64:{}", encoded);
+
+        // When getting content from the base64 URI
+        let result = get_content(base64_uri).await;
+
+        // Then the content should be successfully decoded
+        assert!(result.is_ok());
+        assert_eq!(result.unwrap(), test_content);
+    }
+
+    #[tokio::test]
+    async fn test_get_content_base64_invalid() {
+        // Given an invalid base64 string that cannot be decoded
+        // When attempting to get content from the malformed base64 URI
+        let result = get_content("base64:invalid!@#$").await;
+
+        // Then the operation should fail with a base64 decode error
+        assert!(result.is_err());
+        assert!(result
+            .unwrap_err()
+            .to_string()
+            .contains("Failed to decode base64"));
+    }
+
+    #[tokio::test]
+    async fn test_get_content_base64_invalid_utf8() {
+        // Given a valid base64 string that decodes to invalid UTF-8 bytes
+        let invalid_bytes = vec![0xFF, 0xFE, 0xFD];
+        let encoded = engine::general_purpose::STANDARD.encode(&invalid_bytes);
+        let base64_uri = format!("base64:{}", encoded);
+
+        // When attempting to get content from the base64 URI
+        let result = get_content(base64_uri).await;
+
+        // Then the operation should fail with a UTF-8 validation error
+        assert!(result.is_err());
+        assert!(result.unwrap_err().to_string().contains("not valid UTF-8"));
+    }
+
+    #[tokio::test]
+    async fn test_uri_parsing() {
+        // Given an invalid URI string that cannot be parsed
+        // When attempting to get content from the malformed URI
+        let result = get_content("invalid-uri").await;
+
+        // Then the operation should fail with a URI parsing error
+        assert!(result.is_err());
+        assert!(result.unwrap_err().to_string().contains("Invalid URI"));
+    }
+
+    #[tokio::test]
+    async fn test_http_uri_rejected() {
+        // Given an HTTP URI
+        // When attempting to get content from the HTTP URI
+        let result = get_content("http://example.com/net.toml").await;
+
+        // Then the operation should fail with an unsupported URI scheme error
+        assert!(result.is_err());
+        let error_msg = result.unwrap_err().to_string();
+        assert!(error_msg.contains("Unsupported URI scheme 'http'"));
+        assert!(error_msg.contains("Only file:// and base64: schemes are supported"));
+    }
+
+    #[tokio::test]
+    async fn test_https_uri_rejected() {
```

Comment:
Can you use your awesome `test_case` here and have the one test for the different use cases?

Reply:
Done.